cpp/dolfinx/io/ADIOS2_utils.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2024 Abdullah Mujahid, Jørgen S. Dokken and Garth N. Wells
+//
+// This file is part of DOLFINX (https://www.fenicsproject.org)
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+
+#ifdef HAS_ADIOS2
+
+#include "ADIOS2_utils.h"
+#include <adios2.h>
+#include <mpi.h>
+
+namespace
+{
+std::map<std::string, adios2::Mode> string_to_mode{
+    {"write", adios2::Mode::Write},
+    {"read", adios2::Mode::Read},
+};
+} // namespace
+
+using namespace dolfinx::io;
+
+//-----------------------------------------------------------------------------
+ADIOS2Wrapper::ADIOS2Wrapper(MPI_Comm comm, std::string filename,
```

Comment:
This class should additionally support passing a path to an Adios2 configuration file. This will allow Python users to setup the Adios2 with all of the supported options. Not ideal but it will work.

Reply:
Done.